api/src/main/java/org/openmrs/module/queue/tasks/AutoCloseVisitQueueEntryTask.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.queue.tasks;
+
+import java.util.Collection;
+import java.util.Date;
+
+import lombok.extern.slf4j.Slf4j;
+import org.openmrs.Visit;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.queue.api.QueueEntryService;
+import org.openmrs.module.queue.api.VisitQueueEntryService;
+import org.openmrs.module.queue.model.QueueEntry;
+import org.openmrs.module.queue.model.VisitQueueEntry;
+
+/**
+ * This iterates over all active VisitQueueEntries If the Visit associated with any of these has
+ * ended (non-null stopDatetime), then the QueueEntry associated with it is also ended at the same
+ * datetime as the Visit was stopped.
+ */
+@Slf4j
+public class AutoCloseVisitQueueEntryTask implements Runnable {
+	
+	private static boolean currentlyExecuting = false;
```

Comment:
Sorry for the post-commit review, but if this accessed concurrently (which seems to be the use case) it should be marked as `volatile`. Otherwise, if two threads that start close to one another, they might both see this value as `false` (or alternatively, but less importantly, as `true`).